Ease of use

serpi90 opened the pull request:
- Don't stop stop the UI Process when closing a revived fuel stack
- Extract logging behavior and move defaults to class side, this should make it easier to log / dump stack files.
- Default files moved to class side to be able to reference them easily.
- Comment subclassResponsibility methods, for clarity

Some of the changes are breaking, this should move to next major version.

serpi90 committed:
fix: comment subclassResponsibility methods, for clarity

serpi90 committed:
BREAKING feature: extract logging and move defaults to class side

this should make it easier to log / dump stack files.
- Logs can now be sent to CurrentLogger, a dynamic variable.
- defaults moved to class side to be able to reference them easily.

serpi90 committed:
doc: fix markdown linting

serpi90 committed:
fix: Attempt to fix problems with stdio references

When the logger is used and the image is saved, a reference to stdout/stderr is kept alive.
This would later fail when the image starts again because the reference is invalid.

Remove the class variable keeping the default to try to fix that.

serpi90 committed:
fix: Log file timestamp should include timezone & milliseconds

serpi90 committed:
fix: Dont stop stop the UI Process when closing a revived fuel stack

jvanecek committed:
feat: add new message logging error during execution

jvanecek reviewed (dismissed):
Pretty useful the logger. Might be a product on its own. I just add a new message that logs any error occurred during an execution

gcotelli commented on source/Application-Starter-Tests/DummyApplicationStarterCommandLineHandler.class.st:
```diff
@@ -14,19 +16,37 @@ DummyApplicationStarterCommandLineHandler class >> commandName [
+{ #category : #'private - accessing' }
+DummyApplicationStarterCommandLineHandler class >> defaultLogsPath: aPath [
+
+	memoryLogPasthConfiguredFromTest := aPath
```
Typo memoryLog**Pasth**ConfiguredFromTest -> memoryLogPathConfiguredFromTest

gcotelli commented on source/Application-Starter/ErrorStackSerializer.class.st:
```diff
@@ -7,12 +7,13 @@ Class {
 				newDebugSessionNamed: ( 'Materialized stack: <1s>' expandMacrosWith: anExecutionStack receiver description )
 				startedAt: anExecutionStack )
 		withFullView: true
+	] fork
```
I suggest forking the process with a proper name

gcotelli commented on source/Application-Starter/LeveledLogger.class.st:
```diff
@@ -0,0 +1,110 @@
+]
+
+{ #category : #logging }
+LeveledLogger >> error: anErrorMessage [
```
Why not logError: ?

serpi90 replied:
Because you already are in the context of a logger, and i considered it to be redundant.
But i don't really care if we change it to log<level>

jvanecek commented on source/Application-Starter/ApplicationStarterCommandLineHandler.class.st:
```diff
@@ -139,94 +150,22 @@ ApplicationStarterCommandLineHandler >> isDebugModeEnabled [
-]
-
-{ #category : #'private - logging' }
-ApplicationStarterCommandLineHandler >> logInfo: anErrorMessage during: aBlock [
```
Erasing this messages breaks compatibility and every application starter using it

jvanecek committed:
fix: restore deleted protocol that breaks compatibility

serpi90 commented:
A migration package would be better, with a rewrite rule instead of mantaining the old method forever

serpi90 committed:
refactor: move compatilibility methods to a separate package

Needs migration instructions in the documentation, and possibly a script to migrate the logPrefix methods from instance side to command side

serpi90 committed:
feature: add missing messages for migration

gcotelli committed:
Fix typo in class variable name
Rename Logger methods
Fork the materialization process with a proper name

serpi90 commented on source/Application-Starter-Migration-V1-V2/ApplicationStarterCommandLineHandler.extension.st:
```diff
@@ -0,0 +1,145 @@
+		in: 'v2'
+		transformWith: '`@receiver logError: ̀@message' -> '`CurrentLogger value error: `@message'.
+
+	CurrentLogger value error: anErrorMessage
```
This should have been renamed as `logAsError:` (same for all other message sends to CurrentLogger in this extension)

gcotelli replied:
:+1 I was missing the unloaded packages.

gcotelli committed:
Fix migration package to use the new methods
